src/kvstore.c

```diff
@@ -596,13 +600,15 @@ int kvstoreIteratorGetCurrentDictIndex(kvstoreIterator *kvs_it) {
 dictEntry *kvstoreIteratorNext(kvstoreIterator *kvs_it) {
     dictEntry *de = kvs_it->di.d ? dictNext(&kvs_it->di) : NULL;
     if (!de) { /* No current dict or reached the end of the dictionary. */
+
+        /* Before we move to the next dict, function kvstoreIteratorNextDict()
+         * reset the iter of the previous dict. */
         dict *d = kvstoreIteratorNextDict(kvs_it);
+
         if (!d)
             return NULL;
+
         if (kvs_it->di.d) {
-            /* Before we move to the next dict, reset the iter of the previous dict. */
-            dictIterator *iter = &kvs_it->di;
-            dictResetIterator(iter);
             /* In the safe iterator context, we may delete entries. */
             freeDictIfNeeded(kvs_it->kvs, kvs_it->didx);
```

Comment:
does freeDictIfNeeded will be called twice as well?

Reply:
i agree. @moticless?

Reply:
Agree as well :)

I wasn't sure if it is ok to delete it because its twisted behviour. Note that the wrapping condition is being asked on previous dict, yet `freeDictIfNeeded()` is being called after updating the value of `kvs_it->didx` to point next dict, which for sure not right logic.

This single line was carefully added which made me doubt if there is something that i am missing. After verified that this function make no harm in case called more than once, i decided to leave it as is.

I will remove it. thanks.